content/blog/parsnip-0-1-2/index.Rmd

```diff
@@ -0,0 +1,279 @@
+## Variable importance mode:         none 
+## Splitrule:                        variance 
+## OOB prediction error (MSE):       224771.2 
+## R squared (OOB):                  0.6533301
+```
+
+```r
+ranger(body_mass_g ~ species + island, data = penguins, seed = 1221)
+```
+
+```
+## Ranger result
+## 
+## Call:
+##  ranger(body_mass_g ~ species + island, data = penguins, seed = 1221) 
+## 
+## Type:                             Regression 
+## Number of trees:                  500 
+## Sample size:                      333 
+## Number of independent variables:  2 
+## Mtry:                             1 
+## Target node size:                 5 
+## Variable importance mode:         none 
+## Splitrule:                        variance 
+## OOB prediction error (MSE):       224771.2 
+## R squared (OOB):                  0.6533301
+```
+
+(Note the `Number of independent variables:  2`). 
+
+However, the workflows package does process the data before giving it to the modeling function. In this case, indicators were produced when a formula is used. As a result, instead of two independent variables, the `species` was expanded and five predictor columns are given to the model: 
+
+```r
+library(workflows)
+library(hardhat)
+
+rf_wflow <-
+  workflow() %>%
+  add_model(rf_spec) %>%
+  add_formula(body_mass_g ~ species + island)
+
+rf_wflow %>%
+  fit(data = penguins)
+```
+
+```
+## ══ Workflow [trained] ══════════════════════════════════════════════════════════════════════
+## Preprocessor: Formula
+## Model: rand_forest()
+## 
+## ── Preprocessor ────────────────────────────────────────────────────────────────────────────
+## body_mass_g ~ species + island
+## 
+## ── Model ───────────────────────────────────────────────────────────────────────────────────
+## Ranger result
+## 
+## Call:
+##  ranger::ranger(formula = formula, data = data, seed = ~1221, num.threads = 1, verbose = FALSE) 
+## 
+## Type:                             Regression 
+## Number of trees:                  500 
+## Sample size:                      333 
+## Number of independent variables:  5 
+## Mtry:                             2 
+## Target node size:                 5 
+## Variable importance mode:         none 
+## Splitrule:                        variance 
+## OOB prediction error (MSE):       215619.5 
+## R squared (OOB):                  0.6674451
+```
+
+Not only was the inconsistency of these two interfaces a problem, but `ranger` is very persnickety about  column names and some indicator columns would result in errors (see, for example, [this issue](https://github.com/tidymodels/tune/issues/151)).
+
+The new set of hardhat/workflow/parsnip versions now fix this behavior. In parsnip, each model/engine combination has a recommended set of predictor encoding methods attached to them (including a "leave my data alone" option). These are designed to be consistent with what the underlying model function expects so that there are no inconsistencies. 
```

Comment:
```suggestion
The new set of hardhat/workflow/parsnip versions now fixes this behavior. In parsnip, each model/engine combination has a recommended set of predictor encoding methods attached to them (including a "leave my data alone" option). These are designed to be consistent with what the underlying model function expects so that there are no inconsistencies. 
```